src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy

```diff
@@ -207,16 +206,56 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
      * @return The objdump executable, or null if not found
      */
     File getObjDumpExecutable(String arch) {
-
         try {
-            Abi abi = Abi.getByName(arch)
-            NdkHandler handler = new NdkHandler(rootDir, null, toolchain, "", true)
-            File objDumpPath = new File(handler.getDefaultGccToolchainPath(abi), "bin/" + abi.getGccExecutablePrefix() + "-objdump")
-            return objDumpPath
+            String override = getObjDumpOverride(arch)
+            File objDumpFile
+
+            if (override != null) {
+                objDumpFile = new File(override)
+            } else {
+                objDumpFile = findObjDump(project, arch)
+            }
+
+            if (!objDumpFile.exists() || !objDumpFile.canExecute()) {
+                throw new RuntimeException("Failed to find executable objdump at $objDumpFile")
+            }
+            return objDumpFile
         } catch (Throwable ex) {
             project.logger.error("Error attempting to calculate objdump location: " + ex.message)
         }
-
         return null
     }
+
+    private Object getObjDumpOverride(String arch) {
+        Map<String, String> paths = project.bugsnag.objdumpPaths
+        return paths != null ? paths[arch] : null
+    }
+
+    static File findObjDump(Project project, String arch) {
+        Abi abi = Abi.findByName(arch)
+        String ndkDir = project.android.ndkDirectory
+        String osName = calculateOsName()
+        return calculateObjDumpLocation(ndkDir, abi, osName)
+    }
+
+    static File calculateObjDumpLocation(String ndkDir, Abi abi, String osName) {
+        new File("$ndkDir/toolchains/$abi.toolchainPrefix-4.9/prebuilt/$osName/bin/$abi.objdumpPrefix-objdump")
+    }
+
+    static String calculateOsName() {
+        if (Os.isFamily(Os.FAMILY_MAC)) {
+            return "darwin-x86_64"
+        } else if (Os.isFamily(Os.FAMILY_UNIX)) {
+            return "linux-x86_64"
+        } else if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+            if ("x86" == System.getProperty("os.arch")) { // 32-bit
+                return "windows"
+            } else {
+                return "windows-x86_64"
+            }
+        } else {
+            return "unknown"
```

Comment:
is there something better we can do here? such as raising an exception or logging a warning? Tell people to use the `objdumpPaths` option? I think once you reach this branch and it builds a string that we know won't work, the error message will be confusing or missing enough detail to work out what went wrong